dev/generate_bom.py

```diff
@@ -289,14 +289,25 @@ def __publish_config(self, component, profile_path):
     """
     for profile in os.listdir(profile_path):
       full_profile = os.path.join(profile_path, profile)
+      profile_path = full_profile
```

Comment:
Talked offline, turns out there can be directories in the halconfig `profile_path`, which we'd want to tar up in the future, while treating the toplevel files the same.